src/service.cc

```diff
@@ -168,6 +169,11 @@ Status Service::genotype_sites(const genotyper_config& cfg, const string& sample
         return Status::Failure("bcf_hdr_sync");
     }
 
+    // safeguard against update_genotypes failure from improper header
+    if(bcf_hdr_nsamples(hdr) != samples->size()){
+        return Status::Failure("Mismatch found in number of samples in output bcf header");
+    }
```

Comment:
I think `assert(bcf_hdr_nsamples(hdr) == samples->size());` would be slightly more appropriate than this unconditional runtime check. If it triggers then it implies there's a bug. Unless it's somehow possible for `bcf_add_sample` above to succeed and then for this to trigger?